sdk/core/Microsoft.Azure.Core.Spatial.NewtonsoftJson/src/JsonExtensions.cs

```diff
@@ -0,0 +1,277 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// Defines extension methods for various JSON.NET types that make it easier to implement a custom JsonConverter.
+    /// </summary>
+    internal static class JsonExtensions
+    {
+        /// <summary>
+        /// Asserts that the given JSON reader is positioned on a token with the expected type. Optionally asserts
+        /// that the value of the token matches a given expected value. If any of the assertions fail, this method
+        /// throws a JsonSerializationException. Otherwise, this method attempts to advance the JSON reader to the
+        /// next position.
+        /// </summary>
+        /// <param name="reader">The JSON reader.</param>
+        /// <param name="expectedToken">The JSON token on which the reader is expected to be positioned.</param>
+        /// <param name="expectedValues">Optional; The expected possible values of the current JSON token.</param>
+        public static void ExpectAndAdvance(
+            this JsonReader reader,
+            JsonToken expectedToken,
+            params object[] expectedValues) => ExpectAndAdvance<object>(reader, expectedToken, expectedValues);
```

Comment:
Why return `void` instead of the `object`?

Reply:
This variant only asserts and advances. There is another variant that returns an object of type `T`. This is migrated from track 1. When I work on #15506 I may rewrite all of the migrated track 1 anyway.